[2025-08-13T10:07:23.725+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather_batch_processing.run_weather_batch_script manual__2025-08-13T10:02:21.153659+00:00 [queued]>
[2025-08-13T10:07:23.738+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather_batch_processing.run_weather_batch_script manual__2025-08-13T10:02:21.153659+00:00 [queued]>
[2025-08-13T10:07:23.738+0000] {taskinstance.py:2171} INFO - Starting attempt 2 of 2
[2025-08-13T10:07:23.756+0000] {taskinstance.py:2192} INFO - Executing <Task(PythonOperator): run_weather_batch_script> on 2025-08-13 10:02:21.153659+00:00
[2025-08-13T10:07:23.760+0000] {standard_task_runner.py:60} INFO - Started process 1678 to run task
[2025-08-13T10:07:23.765+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'weather_batch_processing', 'run_weather_batch_script', 'manual__2025-08-13T10:02:21.153659+00:00', '--job-id', '11', '--raw', '--subdir', 'DAGS_FOLDER/weather_batch_dag.py', '--cfg-path', '/tmp/tmpo51ffhff']
[2025-08-13T10:07:23.768+0000] {standard_task_runner.py:88} INFO - Job 11: Subtask run_weather_batch_script
[2025-08-13T10:07:23.843+0000] {task_command.py:423} INFO - Running <TaskInstance: weather_batch_processing.run_weather_batch_script manual__2025-08-13T10:02:21.153659+00:00 [running]> on host b8abf4c23708
[2025-08-13T10:07:23.987+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='weather_batch_processing' AIRFLOW_CTX_TASK_ID='run_weather_batch_script' AIRFLOW_CTX_EXECUTION_DATE='2025-08-13T10:02:21.153659+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-13T10:02:21.153659+00:00'
[2025-08-13T10:07:23.988+0000] {logging_mixin.py:188} INFO - üöÄ Running Python script: /opt/airflow/pyspark_jobs/process_weather_data.py
[2025-08-13T10:07:24.074+0000] {logging_mixin.py:188} INFO - üìÑ STDOUT:
[2025-08-13T10:07:24.074+0000] {logging_mixin.py:188} INFO -  
[2025-08-13T10:07:24.075+0000] {logging_mixin.py:188} INFO - ‚ö†Ô∏è STDERR:
[2025-08-13T10:07:24.075+0000] {logging_mixin.py:188} INFO -  python3: can't open file '/opt/airflow/pyspark_jobs/process_weather_data.py': [Errno 2] No such file or directory
[2025-08-13T10:07:24.075+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/weather_batch_dag.py", line 21, in run_weather_script
    raise RuntimeError(f"Script failed with exit code {result.returncode}")
RuntimeError: Script failed with exit code 2
[2025-08-13T10:07:24.101+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=weather_batch_processing, task_id=run_weather_batch_script, execution_date=20250813T100221, start_date=20250813T100723, end_date=20250813T100724
[2025-08-13T10:07:24.122+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 11 for task run_weather_batch_script (Script failed with exit code 2; 1678)
[2025-08-13T10:07:24.137+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-13T10:07:24.166+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
